express-notes-api/server.js

```diff
@@ -0,0 +1,118 @@
+import express from 'express';
+import { readFile, writeFile } from 'node:fs/promises';
+
+const app = express();
+app.use(express.json());
+
+app.get('/api/notes', async (req, res) => {
+  const array = [];
+  const data = await readJSONData();
+  const notesList = data.notes;
+  for (const entries in notesList) {
+    array.push(notesList[entries]);
+  }
+  res.json(array);
+});
+
+app.get('/api/notes/:id', async (req, res) => {
+  const { id } = req.params;
+  const data = await readJSONData();
+  const notesList = data.notes;
+  if (id <= 0) {
+    res.json({ error: 'Please enter a positive number.' });
+  } else if (notesList[id]) {
+    res.json(notesList[id]);
+  } else {
+    res.status(404).json({ error: `Entry Id: ${id} does not exist.` });
+  }
+});
+
+app.post('/api/notes', async (req, res) => {
+  try {
+    const newEntry = req.body;
+    const data = await readJSONData();
+    if (!newEntry.content) {
+      res.status(400).json({ error: 'Content is a required field.' });
+    } else if (newEntry.content) {
```

Comment:
No need to check newEntry.content again. You already know from line 34 that it's truthy.